hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/HoodieMergeOnReadRDD.scala

```diff
@@ -54,15 +54,20 @@ class HoodieMergeOnReadRDD(@transient sc: SparkContext,
   private val preCombineField = tableState.preCombineField
   private val recordKeyFieldOpt = tableState.recordKeyFieldOpt
   private val payloadProps = if (preCombineField.isDefined) {
-    Some(HoodiePayloadConfig.newBuilder.withPayloadOrderingField(preCombineField.get).build.getProps)
+    val properties = HoodiePayloadConfig.newBuilder
+      .withPayloadOrderingField(preCombineField.get)
+      .withPayloadEventTimeField(preCombineField.get)
```

Comment:
this is a bug if not set?

Reply:
i think yes.
If not set, upsert may get a wrong result if use `DefaultHoodieRecordPaylod`. And in spark3.2, if no `ordering.field` or `event.time.field` found, setting `returnNullIfNotFound` to true does not take effect when call `HoodieAvroUtils.getNestedFieldVal`, like #4169.

Reply:
> setting returnNullIfNotFound to true does not take effect when call HoodieAvroUtils.getNestedFieldVal

this looks like a bug in `getNestedFieldVal()` ? is it fixable in spark 3.2?

Reply:
I think not. But I also have no idea about the exception when set `returnNullIfNotFound` true and call `getNestedFieldVal`. But if `PAYLOAD_ORDERING_FIELD_PROP_KEY` is set automatically when `preCombineField` is set, which we should have done this, it works well.